discretisedfield/field.py

```diff
@@ -3435,152 +3435,224 @@ def _hv_key_dims(self):
             key_dims["vdims"] = hv_key_dim(self.vdims, "")
         return key_dims
 
-    @property
     def fftn(self):
         """Fourier transform.
 
-        Computes 3D FFT for "normal" fields, 2D FFT if the field is sliced.
+        N dimentional discrete FFT of the field.
 
         Returns
         -------
         discretisedfield.Field
+
+            Fourier transformed field.
+
+        Examples
+        --------
+        1. Create a mesh and perform a FFT.
+        >>> import discretisedfield as df
+        >>> mesh = df.Mesh(p1=0, p2=10, cell=2)
+        >>> field = df.Field(mesh, dim=3, value=(1, 2, 3))
+        >>> fft_field = field.fftn()
+        >>> fft_field.nvdim
+        array([3])
+        >>> fft_field.vdims
+        ('ft_x', 'ft_y', 'ft_z')
+
         """
-        mesh = self._fft_mesh()
+        mesh = self.mesh.fftn()
+
+        axes = range(self.mesh.region.ndim)
+        ft = np.fft.fftshift(
+            np.fft.fftn(self.array, axes=axes),
+            axes=axes,
+        )
+
+        if self.vdims is not None:
+            ft_vdims = [f"ft_{vdim}" for vdim in self.vdims]
 
-        values = []
-        for idx in range(self.nvdim):
-            ft = np.fft.fftshift(np.fft.fftn(self.array[..., idx].squeeze()))
-            values.append(ft.reshape(mesh.n))
+            ft_vdim_mapping = {}
+            for vdim, ft_vdim in zip(self.vdims, ft_vdims):
+                if vdim in self.vdim_mapping.keys():
+                    ft_vdim_mapping[ft_vdim] = "k_" + self.vdim_mapping[vdim]
+        else:
+            ft_vdims = None
+            ft_vdim_mapping = None
 
         return self.__class__(
             mesh,
-            nvdim=len(values),
-            value=np.stack(values, axis=3),
-            vdims=self.vdims,
-            # TODO vdim_mapping with fft_mesh dims
+            nvdim=self.nvdim,
+            value=ft,
+            vdims=ft_vdims,
+            unit=self.unit,
+            vdim_mapping=ft_vdim_mapping,
         )
 
-    @property
     def ifftn(self):
         """Inverse Fourier transform.
 
+        N dimentional discrete inverse FFT of the field.
+
         Returns
         -------
         discretisedfield.Field
+
+            Inverse Fourier transformed field.
+
+        Examples
+        --------
+        1. Create a mesh and perform an iFFT.
+        >>> import discretisedfield as df
+        >>> mesh = df.Mesh(p1=0, p2=10, cell=2)
+        >>> field = df.Field(mesh, dim=3, value=(1, 2, 3))
+        >>> ifft_field = field.fftn().ifftn()
+        >>> ifft_field.nvdim
+        array([3])
+        >>> ifft_field.vdims
+        ('x', 'y', 'z')
+
         """
-        mesh = self.mesh.attributes["realspace_mesh"]
-        if self.mesh.attributes["isplane"] and not mesh.attributes["isplane"]:
-            mesh = mesh.plane(self.mesh.region.dims[self.mesh.attributes["planeaxis"]])
+        mesh = self.mesh.ifftn()
+
+        axes = np.arange(self.mesh.region.ndim)
```

Comment:
FFT did use `range`, here you use `np.arange`. Is there any reason for the difference?